### ISSUE TITLE
False positive bugprone-throw-keyword-missing with NSDMI

### ISSUE THREAD

**Issue by carlosgalvezp** [llvm/llvm-project#115055]
Example:

```cpp
struct RegularException
{
    RegularException(int);
};

class Foo
{
 public:
    Foo(){} 

 private:
    RegularException x_{123};
};

```

Gives:
```console
[source>:12:24: warning: suspicious exception object created but not thrown; did you mean 'throw RegularException'? [bugprone-throw-keyword-missing]](javascript:;)
   12 |     RegularException x_{123};
      |                        ^
```

[Godbolt](https://godbolt.org/z/47rsz97hf)
